### ISSUE TITLE
Postgres 9.4.4  process not started automatically by patroni

### ISSUE THREAD

**Issue by sachin2411** [zalando/patroni#895]
I am trying to convert my existing postgres node(with existing data ) into a patroni cluster.
but seems there is a issue below is log
Dec 12 18:04:57 localhost systemd: Configuration file /etc/systemd/system/patroni.service is marked executable. Please remove executable permission bits. Proceeding anyway.
Dec 12 18:04:57 localhost systemd: Configuration file /etc/systemd/system/patroni.service is marked world-writable. Please remove world writability permission bits. Proceeding anyway.
Dec 12 18:04:57 localhost systemd: Started Runners to orchestrate a high-availability PostgreSQL.
Dec 12 18:04:57 localhost patroni: 2018-12-12 18:04:57,713 INFO: Failed to import patroni.dcs.consul
Dec 12 18:04:57 localhost patroni: 2018-12-12 18:04:57,722 INFO: Selected new etcd server http://xxxx:2379
Dec 12 18:04:57 localhost patroni: 2018-12-12 18:04:57,739 WARNING: Postgresql is not running.
Dec 12 18:04:57 localhost patroni: 2018-12-12 18:04:57,739 INFO: Lock owner: None; I am postgresql1
Dec 12 18:04:57 localhost patroni: 2018-12-12 18:04:57,744 INFO: Lock owner: None; I am postgresql1
Dec 12 18:04:57 localhost patroni: 2018-12-12 18:04:57,748 INFO: Lock owner: None; I am postgresql1
Dec 12 18:04:57 localhost patroni: 2018-12-12 18:04:57,750 INFO: starting as a secondary
Dec 12 18:04:57 localhost patroni: 2018-12-12 18:04:57,779 INFO: postmaster pid=1549
Dec 12 18:04:57 localhost patroni: localhost:5432 - no response
Dec 12 18:04:58 localhost patroni: 2018-12-12 18:04:58,787 ERROR: postmaster is not running
Dec 12 18:04:58 localhost patroni: 2018-12-12 18:04:58,820 INFO: Lock owner: None; I am postgresql1
Dec 12 18:04:58 localhost patroni: 2018-12-12 18:04:58,821 INFO: failed to start postgres
below is patroni.yml

[patroni.txt](https://github.com/zalando/patroni/files/2671915/patroni.txt)

I tried using patronictl but it is giving 503 error during restart
Kindly help

**Comment by mashuma** [comment]
These line tell that Patroni is starting postgres, it got no response and that postmaster (postgres process) is not running.
```
Dec 12 18:04:57 localhost patroni: 2018-12-12 18:04:57,750 INFO: starting as a secondary
Dec 12 18:04:57 localhost patroni: 2018-12-12 18:04:57,779 INFO: postmaster pid=1549
Dec 12 18:04:57 localhost patroni: localhost:5432 - no response
Dec 12 18:04:58 localhost patroni: 2018-12-12 18:04:58,787 ERROR: postmaster is not running
```

Please check postgres log and / or post it here.

**Comment by sachin2411** [comment]
I started postgres manually and below are the logs
Dec 12 18:20:58 localhost patroni: 2018-12-12 18:20:58,008 WARNING: Postgresql is not running.
Dec 12 18:20:58 localhost patroni: 2018-12-12 18:20:58,008 INFO: Lock owner: None; I am postgresql1
Dec 12 18:20:58 localhost patroni: 2018-12-12 18:20:58,015 INFO: Lock owner: None; I am postgresql1
Dec 12 18:20:58 localhost patroni: 2018-12-12 18:20:58,019 INFO: Lock owner: None; I am postgresql1
Dec 12 18:20:58 localhost patroni: 2018-12-12 18:20:58,021 INFO: starting as a secondary
Dec 12 18:20:58 localhost patroni: 2018-12-12 18:20:58,022 INFO: closed patroni connection to the postgresql cluster
Dec 12 18:20:58 localhost patroni: 2018-12-12 18:20:58,049 INFO: postmaster pid=3043
Dec 12 18:20:58 localhost patroni: localhost:5432 - accepting connections
Dec 12 18:20:58 localhost patroni: 2018-12-12 18:20:58,063 INFO: Lock owner: None; I am postgresql1
Dec 12 18:20:58 localhost patroni: 2018-12-12 18:20:58,063 INFO: establishing a new patroni connection to the postgres cluster
Dec 12 18:20:58 localhost patroni: 2018-12-12 18:20:58,070 INFO: failed to start postgres
Still same  although now 5432 is accepting connections

**Comment by mashuma** [comment]
Patroni is supposed to handle postgres start, stop etc.
If you have started it manually, Patroni may not be able to orchestrate Postgres properly.

Could you please provide postgres log for the time Patroni was starting the postgres server ?

Or alternatively try this:
1. Stop patroni
2. Stop postgres
3. Start patroni

Check if Postgres was started without issues, by patroni.

**Comment by sachin2411** [comment]
followed the same sequence still same .I cant see any log in pg_log folder when patroni starts.When I start postgres manually through service I can see logs there

**Comment by mashuma** [comment]
Wild guess..

In patroni.conf that you have provided, it has this `data_dir: data/patroni`
Please ensure that path is correct and I'd suggest you to use absolute path instead.

It oculd be Patroni is trying to start postgres in new data dir, and this is why you are not finding logs of patroni starting postgres.

**Comment by CyberDem0n** [comment]
My first question would be, why do you run 9.4.4? Please upgrade to 9.4.20.
And the second question: what OS do you use? Are you trying to start Patroni in LXC container?

**Comment by sachin2411** [comment]
Hello  I can upgrade to 9.4.20 and try yes I use centos 
2018-12-12 19:16:07,438 INFO: Failed to import patroni.dcs.consul
2018-12-12 19:16:07,447 INFO: Selected new etcd server http://10.136.112.225:237                                                                                        9
2018-12-12 19:16:07,492 WARNING: Postgresql is not running.
2018-12-12 19:16:07,492 INFO: Lock owner: None; I am postgresql113
2018-12-12 19:16:07,498 INFO: Lock owner: None; I am postgresql113
2018-12-12 19:16:07,502 INFO: Lock owner: None; I am postgresql113
2018-12-12 19:16:07,513 INFO: starting as a secondary
2018-12-12 19:16:07,563 INFO: postmaster pid=8257
localhost:5432 - no response
localhost:5432 - accepting connections
localhost:5432 - accepting connections
2018-12-12 19:16:08,632 INFO: establishing a new patroni connection to the postg                                                                                        res cluster
2018-12-12 19:16:08,640 WARNING: Could not activate Linux watchdog device: "Can'                                                                                        t open watchdog device: [Errno 2] No such file or directory: '/dev/watchdog'"
2018-12-12 19:16:08,646 INFO: promoted self to leader by acquiring session lock
server promoting
2018-12-12 19:16:08,651 INFO: cleared rewind state after becoming the leader
2018-12-12 19:16:12,722 INFO: Lock owner: postgresql113; I am postgresql113
2018-12-12 19:16:12,724 INFO: Lock owner: postgresql113; I am postgresql113
2018-12-12 19:16:12,765 INFO: no action.  i am the leader with the lock
2018-12-12 19:16:22,722 INFO: Lock owner: postgresql113; I am postgresql113
2018-12-12 19:16:22,729 INFO: no action.  i am the leader with the lock
2018-12-12 19:16:32,722 INFO: Lock owner: postgresql113; I am postgresql113
2018-12-12 19:16:32,727 INFO: no action.  i am the leader with the lock
2018-12-12 19:16:42,722 INFO: Lock owner: postgresql113; I am postgresql113
2018-12-12 19:16:42,733 INFO: no action.  i am the leader with the lock
2018-12-12 19:16:52,722 INFO: Lock owner: postgresql113; I am postgresql113
2018-12-12 19:16:52,734 INFO: no action.  i am the leader with the lock
2018-12-12 19:17:02,722 INFO: Lock owner: postgresql113; I am postgresql113
2018-12-12 19:17:02,744 INFO: no action.  i am the leader with the lock
2018-12-12 19:17:12,722 INFO: Lock owner: postgresql113; I am postgresql113
2018-12-12 19:17:12,743 INFO: no action.  i am the leader with the lock

Above is log after clearing the data /patroni folder but still waning persists

**Comment by CyberDem0n** [comment]
``` 
2018-12-12 19:16:07,492 WARNING: Postgresql is not running.
2018-12-12 19:16:08,640 WARNING: Could not activate Linux watchdog device: "Can' t open watchdog device: [Errno 2] No such file or directory: '/dev/watchdog'"```

Don't pay much attention on this logs. It is ok.

**Comment by ants** [comment]
I think we should silence this log or at least make it an INFO when mode is automatic and error is ENOENT. It is too confusing as it stands.

**Comment by sachin2411** [comment]
hi I was able to start and make cluster work for a blank database and then updating data there .What if I have existing data(multiple schema) in tables in my postgres and want to convert it to a cluster using patroni
1.I tried to start postgres manually
2.Then start patroni
3.Tried to restart cluster node via patronictl .But it threw 503 errorbash-4.2$ patronictl -d etcd://10.136.112.225:2379 restart HAPostgres
below is the log
+------------+-----------------+----------------+------+---------+-----------+
|  Cluster   |      Member     |      Host      | Role |  State  | Lag in MB |
+------------+-----------------+----------------+------+---------+-----------+
| HAPostgres | postgresqlnode1 | xxxxx |      | running |     132.0 |
+------------+-----------------+----------------+------+---------+-----------+
Are you sure you want to restart members postgresqlnode1? [y/N]: y
Restart if the PostgreSQL version is less than provided (e.g. 9.5.2)  []:
When should the restart take place (e.g. 2015-10-01T14:30)  [now]:
Failed: restart for member postgresqlnode1, status code=503, (restart failed)

**Comment by sachin2411** [comment]
Hi I was able to setup cluster for a blank database what about existing postgres data folder(with multiple schemas).How do I go about it

**Comment by CyberDem0n** [comment]
Attaching Patroni to the existing cluster is possible. It is only a matter of the providing correct configuration.

If something doesn't work, please look at Patroni and Postgres logs. Usually they have good traces what is going wrong.

**Comment by sachin2411** [comment]
I attatched  patroni and started the leader.in case of secondary server how should I go about it .I created a new data folder and started patroni below is the error it gives  on secondary

pg_basebackup: could not connect to server: FATAL:  no pg_hba.conf entry for replication connection from host "10.136.107.138", user "replication", SSL off

2018-12-14 18:14:54,237 ERROR: Error when fetching backup: pg_basebackup exited with code=1
2018-12-14 18:14:54,238 ERROR: failed to bootstrap from leader 'postgresqlnodeha1'
2018-12-14 18:14:54,239 INFO: Removing data directory: /data/patroni
2018-12-14 18:14:59,182 INFO: Lock owner: postgresqlnodeha1; I am postgresqlnodeha2
2018-12-14 18:14:59,187 INFO: trying to bootstrap from leader 'postgresqlnodeha1'
pg_basebackup: could not connect to server: FATAL:  no pg_hba.conf entry for replication connection from host "xxxxxx", user "replication", SSL off

Anything I am doing wrong . below is .yml file of secondary
**secondary.yml** 
scope: PostgresHA
#namespace: /service/
name: postgresqlnodeha2

restapi:
  listen: xxxx:8008
  connect_address: xxx:8008
#  certfile: /etc/ssl/certs/ssl-cert-snakeoil.pem
#  keyfile: /etc/ssl/private/ssl-cert-snakeoil.key
#  authentication:
#    username: username
#    password: password

# ctl:
#   insecure: false # Allow connections to SSL sites without certs
#   certfile: /etc/ssl/certs/ssl-cert-snakeoil.pem
#   cacert: /etc/ssl/certs/ssl-cacert-snakeoil.pem

etcd:
  host: xxxx:2379

bootstrap:
  # this section will be written into Etcd:/<namespace>/<scope>/config after initializing new cluster
  # and all other cluster members will use it as a `global configuration`
  dcs:
    ttl: 15
    loop_wait: 10
    retry_timeout: 10
    maximum_lag_on_failover: 1048576
#    master_start_timeout: 300
    #synchronous_mode:false
    #standby_cluster:
      #host: 127.0.0.1
      #port: 1111
      #primary_slot_name: patroni
    postgresql:
      use_pg_rewind: true
#      use_slots: true
      parameters:
#        wal_level: hot_standby
#        hot_standby: "on"
#        wal_keep_segments: 8
#        max_wal_senders: 10
#        max_replication_slots: 10
#        wal_log_hints: "on"
#        archive_mode: "on"
#        archive_timeout: 1800s
#        archive_command: mkdir -p ../wal_archive && test ! -f ../wal_archive/%f && cp %p ../wal_archive/%f
#      recovery_conf:
#        restore_command: cp ../wal_archive/%f %p

  # some desired options for 'initdb'
  initdb:  # Note: It needs to be a list (some options need values, others are switches)
  - encoding: UTF8
  - data-checksums

  pg_hba:  # Add following lines to pg_hba.conf after running 'initdb'
  - host replication replication 127.0.0.1/32 md5
  - host replication replication xxxsecondary/32 md5
  - host replication replication  xxxprimary/32 md5
  - host all all 0.0.0.0/0 md5
#  - hostssl all all 0.0.0.0/0 md5

  # Additional script to be launched after initial cluster creation (will be passed the connection URL as parameter)
# post_init: /usr/local/bin/setup_cluster.sh

  # Some additional users users which needs to be created after initializing new cluster
  users:
    admin:
      password: admin
      options:
        - createrole
        - createdb

postgresql:
  listen: xxx:5432
  connect_address: xxx:5432
  data_dir: /data/patroni
#  bin_dir:
#  config_dir:
  pgpass: /tmp/pgpass1
  authentication:
    replication:
      username: replication
      password: replica
    options:
- replication
    superuser:
      username: postgres
      password: xxxxxxx
  parameters:
    unix_socket_directories: '.'

#watchdog:
#  mode: automatic # Allowed values: off, automatic, required
#  device: /dev/watchdog
#  safety_margin: 5

tags:
    nofailover: false
    noloadbalance: false
    clonefrom: false
    nosync: false

**Comment by CyberDem0n** [comment]
Your config is really weirdly formatted:

```yaml
postgresql:
  listen: xxx:5432
  connect_address: xxx:5432
  data_dir: /data/patroni
#  bin_dir:
#  config_dir:
  pgpass: /tmp/pgpass1
  authentication:
    replication:
      username: replication
      password: replica
    options: # WTH?
- replication # WTH?
    superuser:
      username: postgres
      password: xxxxxxx
```

**Comment by sachin2411** [comment]
Will provide formatted configuration files .This is my appraoch
1.My primary postgres node had existing data.I pointed patroni to this data folder it started successfuly as master
2.On my secondary node also has existing replicated data from primary.I tried starting it via patroni pointing the existing data folder at that time it failed with above logs
Error as below
trying to bootstrap from leader 'postgresqlnodeha1'
pg_basebackup: could not connect to server: FATAL: no pg_hba.conf entry for replication connection from host "xxxxxx", user "replication", SSL off
XXX is the master server(leader on cluster)
Is my approach wrong ?

**Comment by sachin2411** [comment]
I pointed the data directory on secondary to the existing data directory .it gives below error
Critical SystemId mismatch postgresqlnodeha2 belongs to existing cluster namespace all are right

**Comment by CyberDem0n** [comment]
If you get `system ID mismatch`, that means that your data directory wasn't copied from the primary and therefore it will not be able to stream from it.

**Comment by CyberDem0n** [comment]
>it does not synce with MASTER

How did you come to such conclusion?
Logs you are posting here don't prove anything. They are just showing that Patroni is trying to determine whether pg_rewind is necessary. Actually pg_rewind isn't available on 9.4 unless you compile it manually.
